### PR TITLE
Add `timeout` option

### LINKED ISSUE #19: Add `timeout` option
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/element-ready/issues/19)

See https://github.com/sindresorhus/element-ready/issues/8#issuecomment-474732172

> There is a $60.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/element-ready/issues/19).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/element-ready/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by bfred-it**
Closes #19

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -53,6 +53,14 @@ Type: `boolean`<br>
 Default: `true`
 
 Automatically stop checking for the element to be ready after the [DOM ready event](https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event). The promise is then resolved to `undefined`.
+The element that's expected to contain a match.
```
Comment (sindresorhus) [inline readme.md]: ?

File: readme.md [hunk 2]
```diff
@@ -53,6 +53,14 @@ Type: `boolean`<br>
...
+
+##### timeout
+
+Type: `number | false`<br>
+Default: `false`
+
+The time to wait before stopping the search, in milliseconds.
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
Time in milliseconds to wait before stopping the search and resolving the promise to `undefined`.
```

?

File: index.d.ts [hunk 3]
```diff
@@ -8,6 +8,13 @@ declare namespace elementReady {
 		@default document
 		*/
 		readonly target?: Element | Document;
+
+		/**
+		The time to wait before stopping the search, in milliseconds.
+
+		@default false
+		*/
+		readonly timeout?: number | false
```
Comment (sindresorhus) [inline index.d.ts]: Should we really do `false`? How about we default to `Infinity` instead?

File: readme.md (hunk 1 above)
Comment (bfred-it) [reply readme.md]: I don't know either

File: readme.md (hunk 2 above)
Comment (bfred-it) [reply readme.md]: `Time in milliseconds to wait` sounded awkward, maybe it's ok.

How about `Milliseconds to wait`?

Comment (sindresorhus) [reply readme.md]: > How about Milliseconds to wait?

👍

File: index.d.ts (hunk 3 above)
Comment (bfred-it) [reply index.d.ts]: I had initially thought of `0` (which is also falsy).

_Do you think V8 will actually timeout when time reaches infinity? And when it does, do we even? I don't even._

That said, I don't know if `timeout: Infinity` makes sense, but if it's a common pattern: 🤷‍♂️

Comment (sindresorhus) [reply index.d.ts]: We just check for `Infinity` manually and do the same as we would with `false`. Passing `Infinity` to setTimeout makes it trigger right away, yay JS :P

Comment (bfred-it) [reply index.d.ts]: > the same as we would with `false`

`timeout > 0` already filters out `false` so it doesn't need extra checks

**Pull request merged**